src/modules/caen/generators/ReadoutGenerator.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2024 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Generator of artificial CAEN readouts
+///
+//===----------------------------------------------------------------------===//
+// GCOVR_EXCL_START
+
+#include <cstdint>
+#include <modules/caen/generators/ReadoutGenerator.h>
+
+namespace Caen {
+
+ReadoutGenerator::ReadoutGenerator() : ReadoutGeneratorBase(ESSReadout::Parser::DetectorType::BIFROST) {
+  app.add_option("--detector", CaenSettings.Detector,
```

Comment:
This could be moved into the base with the improved DetectorType class I suggested, and then in the detector specific generator code we just check if the DetectorType is supported by this generator or not.

Reply:
Currently I wanted to make sure I was backwards compatible so I didn't want to mess with thebase class